mali: support building against 4.13

In 4.13, __GFP_REPEAT is replaced by __GFP_RETRY_MAYFAIL.

See commit dcda9b04713c ("mm, tree wide: replace __GFP_REPEAT by
__GFP_RETRY_MAYFAIL with more useful semantic")

Signed-off-by: Jonathan Liu <net147@gmail.com>
--- a/src/devicedrv/mali/linux/mali_kernel_linux.h
+++ b/src/devicedrv/mali/linux/mali_kernel_linux.h
@@ -18,11 +18,16 @@ extern "C" {
 #include <linux/cdev.h>     /* character device definitions */
 #include <linux/idr.h>
 #include <linux/rbtree.h>
+#include <linux/version.h>
 #include "mali_kernel_license.h"
 #include "mali_osk_types.h"
 
 extern struct platform_device *mali_platform_device;
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 13, 0)
+#define __GFP_REPEAT __GFP_RETRY_MAYFAIL
+#endif
+
 #ifdef __cplusplus
 }
 #endif
